src/SecTester.Scan/CI/VendorMatcher.cs

```diff
@@ -0,0 +1,96 @@
+using System;
+using System.Collections;
+using System.Linq;
+using System.Text.Json;
+
+namespace SecTester.Scan.CI;
+
+internal class VendorMatcher
+{
+  private readonly IDictionary _environment;
+
+  public VendorMatcher(IDictionary environment)
+  {
+    _environment = environment;
+  }
+
+  public bool MatchEnv(JsonElement element)
+  {
+    return element.ValueKind switch
+    {
+      JsonValueKind.String => HasValue(element),
+      JsonValueKind.Object => MatchObject(element),
+      JsonValueKind.Array => element.EnumerateArray().All(HasValue),
+      _ => false
+    };
+  }
+
+  public bool MatchPr(JsonElement element)
+  {
+    return element.ValueKind switch
+    {
+      JsonValueKind.String => HasValue(element),
+      JsonValueKind.Object => NotEqual(element) || MatchObject(element),
+      _ => false
+    };
+  }
+
+  private bool MatchObject(JsonElement element)
+  {
+    return Any(element) || Includes(element) || HasOwnProperties(element);
+  }
+
+  private bool HasValue(JsonElement element)
+  {
+    return element.ValueKind == JsonValueKind.String && !string.IsNullOrEmpty(element.GetString()) && _environment.Contains(element.GetString()!);
+  }
+
+  private bool HasOwnProperties(JsonElement element)
+  {
+    return element.EnumerateObject().All(x =>
+      x.Value.ValueKind == JsonValueKind.String && _environment.Contains(x.Name) &&
+      x.Value.ValueEquals(_environment[x.Name]?.ToString()));
+  }
+
+  private bool Any(JsonElement element)
+  {
+    var any = element.EnumerateObject().FirstOrDefault(x => x.NameEquals("any"));
+
+    return any.Value.ValueKind == JsonValueKind.Array && any.Value.EnumerateArray().Any(HasValue);
+  }
+
+  private bool Includes(JsonElement element)
+  {
+    return ApplyPredicate(element, "includes", (envVarValue, includesValue) =>
+      envVarValue?.Contains(includesValue) ?? false
+    );
+  }
+
+  private bool NotEqual(JsonElement element)
+  {
+    return ApplyPredicate(element, "ne", (envVarValue, neValue) =>
+      neValue == "false" && !string.IsNullOrEmpty(envVarValue)
+    );
+  }
+
+  private bool ApplyPredicate(JsonElement element, string propertyName, Func<string?, string, bool> predicate)
+  {
+    var env = GetPropertyValue(element, "env");
```

Comment:
The method is too complex. Extract a logic to retrieve env to a separate method.

```csharp
private string? GetEnvValue(JsonElement element)
{
  var env = GetPropertyValue(element, "env");

  if (string.IsNullOrEmpty(env) || !_environment.Contains(env))
  {
    return null;
  }

  return _environment[env]?.ToString();
}
```